sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -815,6 +815,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_METASTORE_PARTITION_LIMIT =
+    buildConf("spark.sql.hive.metastorePartitionLimit")
+      .doc("The maximum number of metastore partitions allowed for a given table. The default " +
+           "value -1 to follow the Hive config (see HiveConf.METASTORE_LIMIT_PARTITION_REQUEST " +
```

Comment:
nit: can we use `hive.metastore.limit.partition.request instead of `HiveConf.METASTORE_LIMIT_PARTITION_REQUEST` here? also do we want to mention this is only used for HMS version 2.1 and up?